src/commonMain/kotlin/org/hisp/dhis/rules/engine/RuleVariableValueMapBuilder.kt

```diff
@@ -133,39 +135,35 @@ internal class RuleVariableValueMapBuilder {
         currentDate: LocalDate,
     ): Map<String, RuleVariableValue> {
         val valueMap: MutableMap<String, RuleVariableValue> = HashMap()
-        val eventDate =
-            ruleEvent.eventDate
-                .toLocalDateTime(TimeZone.currentSystemDefault())
-                .date
-                .toString()
-        valueMap[RuleEngineUtils.ENV_VAR_EVENT_DATE] =
-            RuleVariableValue(
-                RuleValueType.DATE,
-                eventDate,
-                listOf(eventDate),
-                currentDate.toString(),
-            )
-        if (ruleEvent.dueDate != null) {
-            val dueDate = ruleEvent.dueDate
-            valueMap[RuleEngineUtils.ENV_VAR_DUE_DATE] =
-                RuleVariableValue(
-                    RuleValueType.DATE,
-                    dueDate.toString(),
-                    listOf(dueDate.toString()),
-                    currentDate.toString(),
-                )
-        }
-        if (ruleEvent.completedDate != null) {
-            val completedDate = ruleEvent.completedDate
-            valueMap[RuleEngineUtils.ENV_VAR_COMPLETED_DATE] =
+            val eventDate =
+                if (ruleEvent.eventDate < Instant.DISTANT_FUTURE )
+                ruleEvent.eventDate
+                    .toLocalDateTime(TimeZone.currentSystemDefault())
+                    .date
+                    .toString()
+            else null
+            valueMap[RuleEngineUtils.ENV_VAR_EVENT_DATE] =
                 RuleVariableValue(
                     RuleValueType.DATE,
-                    completedDate.toString(),
-                    listOf(completedDate.toString()),
+                    eventDate,
+                    if(eventDate == null) listOf() else listOf(eventDate),
```

Comment:
This line
`if(eventDate == null) listOf() else listOf(eventDate),`
could be written as 
`eventDate?.let { listOf(it) } ?: emptyList(),`

They are equivalent. The first one is more Java-like, which could be clearer and easier to understand. The second one is a bit more concise, but perhaps confusing for those unfamiliar with Kotlin. Up to you.

The same comment would apply to more lines in this document